src/index.js

```diff
@@ -46,6 +46,10 @@ try {
 }
 catch (e) { }
 
+if (DEV && (h("div", null).type === "div")) {
```

Comment:
Ah I missed the getter here: https://github.com/developit/preact-compat/pull/537/files#diff-1fdf421c05c1140f6d71444ea2b27638L60-L68

Maybe it's enough if we just check if `vnode.nodeName` is not present. It should only be there for 8.x.